bin/rails-application-template

```diff
@@ -0,0 +1,139 @@
+    "website": "https://solidus.io/",
+    "repository": "https://github.com/solidusio/solidus-example-app",
+    "logo": "https://solidus.io/assets/images/favicon/favicon.ico",
+    "keywords": [
+      "solidus",
+      "example",
+      "rails",
+      "ecommerce",
+      "ruby",
+      "spree",
+      "cart",
+      "store"
+    ],
+    "addons": [
+      "heroku-postgresql:hobby-dev",
+      "memcachier:dev",
+      "cloudinary:starter"
+    ],
+    "scripts": {
+      "postdeploy": "bin/rails db:migrate && bin/rails db:seed spree_sample:load"
+    },
+    "env": {
+      "ADMIN_EMAIL": {
+        "description": "We will create an admin user with this email.",
+        "value": "admin@example.com"
+      },
+      "ADMIN_PASSWORD": {
+        "description": "We will create an admin user with this password.",
+        "value": "test123"
+      },
+      "DEVISE_SECRET_KEY": {
+        "description": "A secret key for salting user passwords.",
+        "generator": "secret"
+      },
+      "RAILS_LOG_TO_STDOUT": {
+        "value": "true"
+      },
+      "RAILS_SERVE_STATIC_FILES": {
+        "value": "true"
+      },
+      "SECRET_KEY_BASE": {
+        "generator": "secret"
+      }
+    }
+  }
+JSON
+
+file "README.md", <<-MD
+  # Solidus Example App
+
+  [![Deploy](https://www.herokucdn.com/deploy/button.svg)](https://heroku.com/deploy?template=https://github.com/nebulab/solidus-example-app)
+MD
+
+initializer "00_solidus_example.rb", <<-RUBY
+  # Needed so database isn't hit on install
+  Spree::Auth::Config.use_static_preferences!
+
+  # Required to work around sample data sending emails
+  Rails.application.config.action_mailer.raise_delivery_errors = false
+RUBY
+
+initializer "devise.rb", <<-RUBY
+  Devise.secret_key = ENV['DEVISE_SECRET_KEY'] || #{SecureRandom.hex(50).inspect}
+RUBY
+
+initializer "paperclip.rb", <<-RUBY
+  if ENV['CLOUDINARY_URL'] || ENV['HEROKU']
+    require 'paperclip/cloudinary'
+
+    Paperclip::Attachment.default_options[:storage] = :cloudinary
+
+    Spree::Image.attachment_definitions[:attachment].delete(:url)
+    Spree::Image.attachment_definitions[:attachment].delete(:path)
+    Spree::Image.attachment_definitions[:attachment].delete(:default_url)
+    Spree::Image.attachment_definitions[:attachment][:path] = 'spree/products/:id/:style/:filename'
+    Spree::Image.attachment_definitions[:attachment][:cloudinary_url_options] = {
+      default: { secure: true }
+    }
+  end
+RUBY
+
+after_bundle do
+  generate(
+    "spree:install",
+    "--auto-accept",
+    "--user_class=Spree::User",
+    "--enforce_available_locales=true",
+  )
+
+  generate('solidus_auth:install')
+  rails_command "db:create"
+  rails_command "db:migrate"
+
+  git :init
+  git add: "."
+  git commit: %{ -m 'Generate app with solidus-example-app template' }
+
+  if ENV['UPDATE_EXAMPLE_APP_REPO']
+    git remote: %{ add origin git@github.com:solidusio/solidus-example-app.git }
+    git push: %{ --force --set-upstream origin master }
```

Comment:
I wasn't sure about using `--force` here without asking the user for confirmation, but I talked myself into being okay with it.

Reply:
The idea is to have something that can be automated and requires as few interaction as possible. That's only for the chore of updating the example app after a release of course, it doesn't happen when used as an app template.

Reply:
Yes, and worst case scenario we just regenerate the example app. The repo we are pushing to will not contain anything we would be afraid to lose.